docs/proposals/secret-manager-store-api.md

```diff
@@ -0,0 +1,69 @@
+# Secrets Manager Store Proposal
+
+**Author**: Emmanuel Bakare (@emmanuel-kubermatic)
+
+**Status**: Draft proposal; prototype in progress.
+
+## Goals
+The proposed fix for the issue [6671](https://github.com/kubermatic/kubermatic/issues/6671) 
+
+Adds support for vault related read access within the kubermatic source
+
+## Non-Goals
+Adds support for vault access and a central store for accessing data
+
+## Motivation and Background
+
+Currently, there's a user ssh agent which manages credentials via fsnotify events and kubernetes secrets.
+
+This approach is great but adds a limitation where the end user has to manually copy credentials to ssh secrets or add them via the API.
+
+Despite the ease of using that, there are some limitations:
+1. Duplicate credentials across their environments.
+2. We can have other secret managers in the future asides Vault 
+
+The approach is to simplify the use / integration of separate secret manager APIs through easy interfaces.
+
+Also, there could be more than 1 secrets manager in the future
+e.g AWS Secrets Manager, Google Secret Manager etc
+
+## Implementation
+
+Define structs and interfaces for other secret managers
+
+we would have interfaces like such
+
+```go
+package sm
+
+type Store interface {
+    write() error
+    read(key string) error
+    init(config interface{}) error
+}
+
+type K8SSecretStore struct{
+	Store
+	// other configs
+}
+```
+
+The approach would this can affect the way we reconcile secrets, even  on Kubernetes as API access to the secrets store could be modified to use those interfaces.
```

Comment:
Similar to my previous comment, where do we want to introduce this abstraction? It's not really clear to me.